File: w0_and_1/pset_0_and_1/mario-less/mario.c
// Mario's pyramid //
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Ask height until answer is within 1 to 8 inclusive //
    int n;
    do
    {
        n = get_int("Enter height from 1 to 8: ");
    }
    while (n < 1 || n > 8);

    // Repeat print n lines //
    for (int i = 0; i < n; i++)
    {
        // Repeat print n-1 amount of spaces for every line (every iteration of a parent loop) //
        for (int k = n - 1; k > i; k--)
        {
            printf(" ");
        }

        // Repeat print # for every line (every iteration of a parent loop) //
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: CS50x/w6/sentimental-mario-more/mario.py
from cs50 import get_int

while True:
    height = get_int("Enter height: ")
    if height > 0 and height < 9:
        break
# For every line
for i in range(height):
    # Spaces for every char in this line. Number of spaces required is associated with height-1.
    for k in range(height - 1):
        print(" ", end="")
    # Reduce amount of spaces for future line
    height -= 1
    # i is 0, so one do loop ine time now, but as i increments with next iteration loop will run one more time with every new iteration.
    for j in range(i + 1):
        print("#", end="")
    print("  ", end="")
    # Explanation from line 12 applies here as well
    for l in range(i + 1):
        print("#", end="")
    print()